### DISCUSSION TITLE
need some help packaging dependencies

### DISCUSSION THREAD

**Discussion by me6262** [bailuk/java-gtk#20] [Q&A]
My app needs to be used on windows, and multiple versions of linux. Namely with version differences between something like opensuse tumbleweed and pop_os, which both have different versions of libadwaita and gtk4. The solution I thought of was to bundle the gtk/libadwaita binaries with the app, but I have no idea what directories to put them in, or if there is some special process to obtaining the binaries. what would be the best way to do these things?

**Comment by bailuk** [comment]
To support different Linux Distributions you could build a Flatpak. I've managed to create a Flatpak for a simple [java-gtk application](https://github.com/bailuk/gtk-meteo). More info in the [README.md here](https://github.com/bailuk/gtk-meteo/tree/main/flatpak) .

**Reply by me6262** [reply]
that was likely what i was leaning  towards on the linux side, however, how would i do it on windows since flatpaks arent a thing on there?

**Reply by me6262** [reply]
but nonetheless, thank you for the great resources for the flatpak. i havent been able to find many good resources on google for this stuff so your help is greatly appreciated

**Comment by me6262** [comment]
for windows, should i just put the corresponding .dll files in the directories that JNA says it looks for them?

**Comment by me6262** [comment]
i figured it out, and now my new app will be ready for my robotics team to use as well as any others that choose to use it, thank you for this great library. i needed it in java so that i could use reflection to interact with our java codebase for our robot.
